src/GovUk.Education.ExploreEducationStatistics.Content.Api.Tests/Controllers/DataSetsControllerTests.cs

```diff
@@ -0,0 +1,366 @@
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(-1)]
+        public async Task PageTooSmall_ReturnsBadRequest(int page)
+        {
+            var query = new DataSetsListRequest(
+                Page: page
+            );
+
+            var client = BuildApp().CreateClient();
+
+            var response = await ListDataSets(client, query);
+
+            response.AssertBadRequest(expectedErrorKey: "Page",
+                expectedErrorMessage: "'Page' must be greater than or equal to '1'.");
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(-1)]
+        public async Task PageSizeTooSmall_ReturnsBadRequest(int pageSize)
+        {
+            var query = new DataSetsListRequest(
+                PageSize: pageSize
+            );
+
+            var client = BuildApp().CreateClient();
+
+            var response = await ListDataSets(client, query);
+
+            response.AssertBadRequest(expectedErrorKey: "PageSize",
+                expectedErrorMessage: "'Page Size' must be greater than or equal to '1'.");
+        }
+
+        [Theory]
+        [InlineData(41)]
+        public async Task PageSizeTooBig_ReturnsBadRequest(int pageSize)
+        {
+            var query = new DataSetsListRequest(
+                PageSize: pageSize
+            );
+
+            var client = BuildApp().CreateClient();
+
+            var response = await ListDataSets(client, query);
+
+            response.AssertBadRequest(expectedErrorKey: "PageSize",
+                expectedErrorMessage: "'Page Size' must be less than or equal to '40'.");
+        }
+
+        [Theory]
+        [InlineData("a")]
+        [InlineData("aa")]
+        public async Task SearchTermTooSmall_ReturnsBadRequest(string searchTerm)
+        {
+            var query = new DataSetsListRequest(
+                SearchTerm: searchTerm
+            );
+
+            var client = BuildApp().CreateClient();
+
+            var response = await ListDataSets(client, query);
+
+            MockUtils.VerifyAllMocks(MemoryCacheService);
+
+            response.AssertBadRequest(expectedErrorKey: "SearchTerm",
+                expectedErrorMessage:
+                $"The length of 'Search Term' must be at least 3 characters. You entered {searchTerm.Length} characters.");
+        }
+
+        [Fact]
+        public async Task OrderByNaturalWithoutReleaseId_ReturnsBadRequest()
+        {
+            var query = new DataSetsListRequest(
+                OrderBy: DataSetsListRequestOrderBy.Natural
+            );
+
+            var client = BuildApp().CreateClient();
+
+            var response = await ListDataSets(client, query);
+
+            MockUtils.VerifyAllMocks(MemoryCacheService);
+
+            response.AssertBadRequest(expectedErrorKey: "ReleaseId",
+                expectedErrorMessage: "'Release Id' must not be empty.");
+        }
+
+        [Fact]
+        public async Task OrderByRelevanceWithoutSearchTerm_ReturnsBadRequest()
```

Comment:
Same as before, I didn't feel the need to provide a success case because the success case should already be covered by the following tests.

```csharp
OrderByTests.OrderByRelevanceAscending_Success()
OrderByTests.OrderByRelevanceDescending_Success()
OrderByTests.OrderByRelevanceWithoutSortOrder_DefaultsToDescending()
```

There's also a couple of other tests  where it should also default to using Relevance as the sort order because they provide a search term without an explicit order by.

```csharp
FilterTests.FilterBySearchTerm_Success()
FilterTests.FilterBySearchTermWhereTermIsNotFound_ReturnsEmpty()

Happy to add in dedicated success cases if you think it's necessary.

Reply:
Update: Added success cases in ef7d583a15d99134500966305da5be3309c2eaa0.

Reply:
Ta